src/shared/Microsoft.Git.CredentialManager.Tests/Interop/U8StringConverterTests.cs

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+using Microsoft.Git.CredentialManager.Interop;
+using Xunit;
+
+namespace Microsoft.Git.CredentialManager.Tests.Interop
+{
+    public class U8StringConverterTests
+    {
+        /*
+         * Grinning face with squinting eyes and sweat drop
+         *   😅
+         *   Codepoint : U+1F605
+         *   UTF-16    : 0xD83D 0xDE05
+         *   UTF-8     : 0xF0 0x9F 0x98 0x85
+         *
+         * Latin small letter dotless I
+         *   ı
+         *   Codepoint : U+0131
+         *   UTF-16    : 0x0131
+         *   UTF-8     : 0xC4 0xB1
+         *
+         * Greek capital letter omega
+         *   Ω
+         *   Codepoint : U+03A9
+         *   UTF-16    : 0x03A9
+         *   UTF-8     : 0xCE 0xA9
+         *
+         * Snowman without snow
+         *   ⛄
+         *   Codepoint : U+26C4
+         *   UTF-16    : 0x26C4
+         *   UTF-8     : 0xE2 0x9B 0x84
+         */
+
+        // "Unicode 😅 ıs awesome! Ω ⛄";
+        private const string ComplexString = "Unicode \uD83D\uDE05 \u0131s awesome! \u03A9 \u26C4";
+        private static readonly byte[] ComplexUtf8 =
+        {
+            (byte)'U', (byte)'n', (byte)'i', (byte)'c', (byte)'o', (byte)'d', (byte)'e', (byte)' ',
+            (byte)'\u00F0', (byte)'\u009F', (byte)'\u0098', (byte)'\u0085', (byte)' ',
+            (byte)'\u00C4', (byte)'\u00B1', (byte)'s', (byte)' ',
+            (byte)'a', (byte)'w', (byte)'e', (byte)'s', (byte)'o', (byte)'m', (byte)'e', (byte)'!', (byte)' ',
+            (byte)'\u00CE', (byte)'\u00A9', (byte)' ',
+            (byte)'\u00E2', (byte)'\u009B', (byte)'\u0084', (byte)'\0'
+        };
+
+        private const string SimpleString = "Hello, World!";
+        private static readonly byte[] SimpleUtf8 =
+        {
+            (byte)'H', (byte)'e', (byte)'l', (byte)'l', (byte)'o', (byte)',', (byte)' ',
+            (byte)'W', (byte)'o', (byte)'r', (byte)'l', (byte)'d', (byte)'!', (byte)'\0'
+        };
+
+        [Fact]
+        public void U8StringConverter_ToNative_SimpleString_ReturnsExpectedBytes()
+        {
+            unsafe
+            {
+                byte* actual = (byte*) U8StringConverter.ToNative(SimpleString);
+
+                AssertCStringEqual(SimpleUtf8,  actual);
+            }
+        }
+
+        [Fact]
+        public void U8StringConverter_ToNative_ComplexString_ReturnsExpectedBytes()
+        {
+            unsafe
+            {
+                byte* actual = (byte*) U8StringConverter.ToNative(ComplexString);
+
+                AssertCStringEqual(ComplexUtf8,  actual);
+            }
+        }
+
+        [Fact]
+        public void U8StringConverter_ToManaged_SimpleString_ReturnsExpectedString()
```

Comment:
Is it work validating the behavior when you send in U8StringConverter.ToManager(null) or U8StringConverter.ToNative(null)?

Reply:
Not much _work_, and definitely _worth_ it! Adding...